toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/AuthenticationActivityExternalBrowserTest.kt

```diff
@@ -0,0 +1,144 @@
+import android.net.Uri
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasAction
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasData
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasExtraWithKey
+import androidx.test.core.app.ActivityScenario
+import com.google.common.truth.Truth.assertThat
+import io.mockk.every
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import org.hamcrest.CoreMatchers.allOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Instrumentation test to verify [AuthenticationActivity]'s behavior.
+ *
+ * @since 300.0.0
+ */
+@RunWith(AndroidJUnit4::class)
+class AuthenticationActivityExternalBrowserTest {
+
+    @Before
+    fun setUp() {
+        Intents.init()
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+        Intents.release()
+    }
+
+    /**
+     * Given [AuthenticationActivity] with a browser that supports Custom Tabs
+     * When [AuthenticationActivity] starts
+     * Then an intent should be fired with ACTION_VIEW, simulating a Custom Tabs launch
+     * And the intent contains the Custom Tabs session extra key
+     * @since 300.0.0
+     */
+    @Test
+    fun launchesCustomTabsWhenSupported() {
+        // Mock the top-level extension so it returns "com.android.chrome" (supports Custom Tabs)
+        mockkStatic("com.arcgismaps.toolkit.authentication.ExtensionsKt")
+        every { any<android.content.Context>().canDefaultBrowserLaunchCustomTabs() } returns true
+
+        // Define the authorize URL to be used in the intent
+        val authorizeUrl = "https://example.com/auth"
+        val intent = Intent(ApplicationProvider.getApplicationContext(), AuthenticationActivity::class.java).apply {
+            putExtra(KEY_INTENT_EXTRA_URL, authorizeUrl)
+        }
+        // Launch the AuthenticationActivity with the intent
+        ActivityScenario.launch<AuthenticationActivity>(intent).use {
+            intended(
+                allOf(
+                    // Verify that the launched intent is a Custom Tabs intent with the expected properties
+                    hasAction(Intent.ACTION_VIEW),
+                    hasData(authorizeUrl),
+                    // Custom Tabs adds android.support.customtabs.extra.SESSION
+                    hasExtraWithKey("android.support.customtabs.extra.SESSION")
+
+                )
+            )
+        }
+    }
+
+    /**
+     * Given [AuthenticationActivity] is launched with a default browser that does not support Custom Tabs
+     * When [AuthenticationActivity] starts
+     * Then the activity finishes with RESULT_CODE_CANCELED and includes an exception message in the result data
+     * @since 300.0.0
+     */
```

Comment:
Would the following be a better description for this test?
```suggestion
    /**
     * Given a device with default browser that does not support Custom Tabs
     * When [AuthenticationActivity] receives an intent for OAuth sign-in
     * Then the activity finishes with RESULT_CODE_CANCELED and includes an exception message in the result data
     * @since 300.0.0
     */
```